### PR TITLE
Test that remote invites are correctly pushed

### PR DESCRIPTION
**Pull request by erikjohnston**
Tests for https://github.com/matrix-org/synapse/pull/15820

### CODE REVIEW

**Comment by erikjohnston** [comment]
I don't understand why there are new failures for Dendrite?

File: tests/61push/01message-pushed.pl [hunk 1]
```diff
@@ -630,3 +630,103 @@ sub check_received_push_with_name
          );
       });
    };
+
+test "Invites over federation are correctly pushed",
```
Comment (MadLittleMods) [inline tests/61push/01message-pushed.pl:634]: Are we making more Sytest tests because there isn't a good way to test push gateway stuff in Complement?

Comment (MadLittleMods) [inline tests/61push/01message-pushed.pl:634]:
```suggestion
test "Invites over federation are correctly pushed (without room_name)",
```

File: tests/61push/01message-pushed.pl [hunk 2]
```diff
@@ -630,3 +630,103 @@ sub check_received_push_with_name
...
+
+         assert_eq( $notification->{room_id}, $room_id, "room_id");
+         assert_eq( $notification->{sender}, $charlie->user_id, "sender");
+
+         Future->done(1);
+      })
+   };
+
+
+test "Invites over federation are correctly pushed with name",
```
Comment (MadLittleMods) [inline tests/61push/01message-pushed.pl:683]:
```suggestion
test "Invites over federation are correctly pushed (with room_name)",
```

**Review by MadLittleMods — APPROVED** [review]
Looks ok, not sure I'd pick up on any Perl intricacies/pitfalls if there was any

**Pull request merged**